google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/hints/Hints.java

```diff
@@ -0,0 +1,515 @@
+ * Root<Singer> root = cr.from(Singer.class);
+ * root.join("albums", JoinType.LEFT);
+ * cr.select(root);
+ * Query<Singer> query = session.createQuery(cr)
+ *     .addQueryHint(
+ *         Hints.forceIndexFrom("singer", "idx_singer_active", ReplaceMode.ALL).toQueryHint())
+ *     .addQueryHint(
+ *         Hints.forceIndexJoin("album", "idx_album_title", ReplaceMode.ALL).toQueryHint());
+ * }</pre>
+ *
+ * <p>Usage with {@link jakarta.persistence.TypedQuery}:
+ *
+ * <pre>{@code
+ * CriteriaBuilder cb = entityManager.getCriteriaBuilder();
+ * CriteriaQuery<Singer> cr = cb.createQuery(Singer.class);
+ * Root<Singer> root = cr.from(Singer.class);
+ * root.join("albums", JoinType.LEFT);
+ * cr.select(root);
+ * TypedQuery<Singer> query = entityManager.createQuery(cr)
+ *     .setHint(AvailableHints.HINT_COMMENT,
+ *         Hints.forceIndexFrom("singer", "idx_singer_active", ReplaceMode.ALl)
+ *             .combine(Hints.forceIndexJoin("album", "idx_album_title", ReplaceMode.ALL))
+ *             .toComment());
+ * }</pre>
+ *
+ * <p>Usage with the {@link jakarta.persistence.QueryHint} annotation:
+ *
+ * <pre>{@code
+ * // The hint value used here is generated by calling the method:
+ * // Hints.forceIndexFrom("singer", "idx_singer_active").toComment()
+ * @QueryHints({@QueryHint(name = AvailableHints.HINT_COMMENT, value = "{\n"
+ *       + "  \"spanner_replacements\": [\n"
+ *       + "    {\n"
+ *       + "      \"regex\": \" from singer \",\n"
+ *       + "      \"replacement\": \" from singer @{FORCE_INDEX=idx_singer_active} \",\n"
+ *       + "      \"replace_mode\": \"ALL\"\n"
+ *       + "    }\n"
+ *       + "  ]\n"
+ *       + "}")})
+ * List<Singer> findByActive(boolean active);
+ * }</pre>
+ */
+public class Hints {
+  /** Possible values for the LOCK_SCANNED_RANGES hint. */
+  public enum LockScannedRanges {
+    EXCLUSIVE,
+    SHARED,
+  }
+
+  /** Possible values for the SCAN_METHOD table hint. */
+  public enum ScanMethod {
+    AUTO,
+    BATCH,
+    ROW,
+  }
+
+  /** Possible values for the INDEX_STRATEGY table hint. */
+  public enum IndexStrategy {
+    FORCE_INDEX_UNION,
+  }
+
+  /** Possible values for the JOIN_METHOD hint. */
+  public enum JoinMethod {
+    HASH_JOIN,
+    APPLY_JOIN,
+    MERGE_JOIN,
+    PUSH_BROADCAST_HASH_JOIN,
+  }
+
+  /** Possible values for the HASH_JOIN_BUILD_SIDE hint. */
+  public enum HashJoinBuildSide {
+    BUILD_LEFT,
+    BUILD_RIGHT,
+  }
+
+  /** Possible values for the HASH_JOIN_EXECUTION hint. */
+  public enum HashJoinExecution {
+    MULTI_PASS,
+    ONE_PASS,
+  }
+
+  /**
+   * Creates a hint that adds @{USE_ADDITIONAL_PARALLELISM=value} to the statement.
+   *
+   * @param value the hint value
+   * @return a hint that can be added as a comment or query hint to a Hibernate statement
+   */
+  public static ReplaceQueryPartsHint useAdditionalParallelism(boolean value) {
+    return ReplaceQueryPartsHint.of(
+        "(?i)(?:^|\\s)(select|insert|update|delete)",
```

Comment:
Can we define this string as a constant?
Also can we define a private function which takes 2 args ("USE_ADDITIONAL_PARALLELISM" and value) so that every other function can use that?

Reply:
Done